iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -60,14 +73,17 @@ public override async Task SendTelemetryAsync(TelemetryMessage message, Cancella
 
             try
             {
+                await VerifyIsOpenAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
I've added a pre-check operation to each API call - `VerifyIsOpenAsync()` once outside the retry block.

If the verification that the client is open is always linked with the retry block, then closing a client and then calling `SendTelemetryAsync()` will result in the SDK complaining that the linked cancellation token source (the one that signals in-progress operations for closure) has been canceled already, instead of complaining that the client is closed.

Reply:
Can be modified further in the "separate-delegating-handlers" PR.